lib/manageiq/appliance_console/postgres_admin.rb

```diff
@@ -320,6 +331,84 @@ class << self
     ensure
       File.delete(error_path) if File.exist?(error_path)
     end
+
+    private_class_method def self.prepare_restore(backup_type, dbname)
+      if application_connections?
+        message = "Database restore failed. Shut down all evmserverd processes before attempting a database restore"
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+
+      disable_replication(dbname)
+
+      conn_count = connection_count(backup_type, dbname)
+      if conn_count > 1
+        message = "Database restore failed. #{conn_count - 1} connections remain to the database."
+        ManageIQ::ApplianceConsole.logger.error(message)
+        raise message
+      end
+    end
+
+    private_class_method def self.application_connections?
+      result = [{"count" => 0}]
+
+      with_pg_connection do |conn|
+        result = conn.exec("SELECT COUNT(pid) FROM pg_stat_activity WHERE application_name LIKE '%MIQ%'")
+      end
+
+      result[0]["count"].to_i > 0
+    end
+
+    private_class_method def self.disable_replication(dbname)
+      require 'pg/logical_replication'
+
+      with_pg_connection do |conn|
+        pglogical = PG::LogicalReplication::Client.new(conn)
+
+        if pglogical.subscriber?
```

Comment:
If it makes sense, I'm ok with pushing down any of this logic into the pg/logical_replication gem.  This could also be done in a follow-up

Reply:
Just because of the delay I have had trying to get this PR, I am going to punt on doing this now and handle in a follow up.

So we don't loose track of it, I will open up an issue that can be triaged later, and we can determine if it makes sense then to push this into the gem.

Reply:
https://github.com/ManageIQ/manageiq-appliance_console/issues/162